.github/workflows/plugin-upgrade-test.yml

```diff
@@ -16,7 +16,23 @@ concurrency:
 
 jobs:
   build-plugin:
-    uses: ./.github/workflows/build.yml
+    name: Build Plugin
+    runs-on: ubuntu-22.04
+    outputs:
+      artifact_name: ${{ steps.build_plugin.outputs.artifact_name }}
+      changelog_diff: ${{ steps.changelog_diffs.outputs.changelog_diff }}
+    steps:
+      - name: Checkout Source Code
+        uses: actions/checkout@v4
+
+      - name: Build Plugin
+        id: build_plugin
+        uses: ./.github/workflows/plugin-build
+
+      - name: Check if this is only a changelog PR
+        id: changelog_diffs
+        uses: ./.github/workflows/check-changelog-pr
```

Comment:
1. I see that we are not using the `check-changelog-pr` output in this file
2. A small suggestion, maybe we should first check if this is a change-log PR and then run the build process?
If this is just a change-log PR there is no need to build the plugin, right?
This could save us some time.
WDYT?